### PR TITLE
Upgrade libraries and environment versions to fix building/releasing.

### PR DESCRIPTION
**Pull request by shayancanonical**
# Issue
Release to charmhub action is [failing](https://github.com/canonical/mysql-k8s-operator/actions/runs/3256480064/jobs/5347273232#step:4:63) due to a change in the way resources are uploaded to charmhub.

# Solution
Upgrade the charming action to v2.1.1 which should resolve the issue

# Release Notes
Upgrade charming actions to v2.1.1 to resolve upload to charmhub issue

### CODE REVIEW

**Review by marcoppenheimer — APPROVED** [review]

**Review by welpaolo — APPROVED** [review]

**Review by MiaAltieri — APPROVED** [review]

**Review by paulomach — APPROVED** [review]

**Comment by taurus-forever** [comment]
@shayancanonical can you please charming actions to 2.1.1. Tnx!

P.S. I have re-triggered integration test as error in PR was different to one I see local on charm deploy:
```
unit-mysql-router-k8s-0: 12:31:39 DEBUG unit.mysql-router-k8s/0.juju-log Emitting Juju event leader_elected.
unit-mysql-router-k8s-0: 12:31:39 DEBUG unit.mysql-router-k8s/0.juju-log HTTP Request: DELETE https://10.96.0.1/api/v1/namespaces/testin
g/services/mysql-router-k8s "HTTP/1.1 403 Forbidden"
unit-mysql-router-k8s-0: 12:31:39 ERROR unit.mysql-router-k8s/0.juju-log Failed to delete k8s service
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-router-k8s-0/charm/venv/lightkube/core/generic_client.py", line 188, in raise_for_status
    resp.raise_for_status()
  File "/var/lib/juju/agents/unit-mysql-router-k8s-0/charm/venv/httpx/_models.py", line 736, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Client error '403 Forbidden' for url 'https://10.96.0.1/api/v1/namespaces/testing/services/mysql-router-k8s'
For more information check: https://httpstatuses.com/403
```

Could be a new ticket to fix... WIP.

**Review by WRFitch — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]
Let's merge this ASAP to unblock all other reviews blocked by flake8 6.0.0 issue.

**Review by WRFitch — APPROVED** [review]

File: .github/workflows/release.yaml
Comment (taurus-forever) [reply .github/workflows/release.yaml:33]: I would like to run 22.04 (which is ubuntu-latest at the moment), but GH is not happy about it (see the commit message):
https://github.com/canonical/mysql-router-k8s-operator/pull/9/commits/1e72239f1e205a4b6bdee571a09a138ac4121873

The link to failing action: https://github.com/canonical/mysql-router-k8s-operator/actions/runs/3830362423/jobs/6518485060
Good ideas are welcome!

Comment (taurus-forever) [reply .github/workflows/release.yaml:33]: I forget to mention, all other CI configs are fine with 22.04 (ubuntu-latest), but charming-actions/upload-charm.

After the second look I noticed '--destructive-mode' in https://github.com/canonical/mysql-router-k8s-operator/actions/runs/3830362423/jobs/6518485060

> --destructive-mode:  Pack charm using current host ...

So, the host MUST match build-on in charmcraft.yaml. Resolved?

Comment (taurus-forever) [reply .github/workflows/release.yaml:33]: Build-on=22.04 and run-on=20.04 might be an option (with some new and unpredictable results).
At the moment we need to update 20.04 version to test Mykola's refactoring without new issues of cross-building.

Shayan recommended to use mysql-bundle approach (call charmcraft manually without '--destructive-mode'): https://github.com/canonical/mysql-bundle/blob/main/Makefile#L21

The good question is: why '--destructive-mode' is mandatory for 'charming-actions'? Can we make it optional?
It will increase usability for us as we want to provide several run-on versions for subordinate charms.

Comment (jnsgruk) [reply .github/workflows/release.yaml:33]: I believe it's used to speed things up. Destructive mode means "don't use a LXD container, just run the build on this host" - which is fine in almost all cases in a CI environment where you don't care too much for the long term integrity of the host (because there is no long-term ;-))

@simskij may be able to give more background, and perhaps we do need to consider allowing this to be optional.

For now, while it's a workaround, I don't see a reason why we need to test/maintain `runs-on==20.04`, we should really be targeting 22.04 in the `charmcraft.yaml`:

```
type: charm
bases:
  - build-on:
      - name: "ubuntu"
        channel: "22.04"
    run-on:
      - name: "ubuntu"
        channel: "22.04"
parts:
  charm:
    charm-binary-python-packages:
      - mysql-connector-python==8.0.30
```

Comment (taurus-forever) [reply .github/workflows/release.yaml]: Updated build-on/run-on to 22.04.

Comment (taurus-forever) [reply .github/workflows/release.yaml]: Reported `upload-charm` action improvement as https://github.com/canonical/charming-actions/issues/80

**Pull request merged**
